server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java

```diff
@@ -270,6 +272,7 @@ public boolean authorizeRequest(RequestDetails requestDetails) {
       return false;
     }
     AccessDecision outcome = checkAuthorization(requestDetails);
+    preprocessRequest(servletDetails, outcome);
```

Comment:
nit: I think you can use `requestDetails` and change the first argument of this method to be the more generic `RequestDetails` instead of `ServletRequestDetails`.